### DISCUSSION TITLE
How can the Vercel GitHub App create a repository without my permission?

### DISCUSSION THREAD

**Discussion by ipselon** [vercel/community#1756] [Help]
### Summary

Hello guys! 

I'm a bit surprised.

How is it possible for the Vercel GitHub App to create a new repository on my GitHub account, even though I only gave it permission to access one of my existing repositories?

Also, I found out that the GitHub App doesn't have access to an API that can create new repositories on a personal account.

Even though the Vercel GitHub App acts on my behalf, it didn't ask for permission to access the 'repo' scope when requesting a personal access token.

### Example

https://vercel.com/new/clone?repository-url=https%3A%2F%2Fgithub.com%2Favitorio%2Foutstatic%2Ftree%2Fmain%2Fexamples%2Fblog&env=OST_GITHUB_ID,OST_GITHUB_SECRET,OST_TOKEN_SECRET&project-name=outstatic-blog&repo-name=outstatic-blog&demo-title=Outstatic%20Blog%20Demo&demo-description=A%20statically%20generated%20blog%20example%20using%20Outstatic&demo-url=https%3A%2F%2Foutstatic-example-blog.vercel.app%2F&demo-image=https%3A%2F%2Foutstatic.com%2Fimages%2Foutstatic-demo.png&envDescription=API%20Keys%20needed%20for%20installation&envLink=https%3A%2F%2Foutstatic.com%2Fdocs%2Fenvironment-variables

**Comment by ustun-ed** [comment]
@ipselon I was surprised by this too. Turns out "it can act on your behalf" which is very broad and dangerous. Wondering if it is a limitation on GitHub side to provide scoped permissions, or Vercel is asking too much.

**Comment by dirk-bester** [comment]
So are these permissions locked down by you to begin with? When I followed GitHub "secure your stuff" directions there was quite a large set of options, including various repo access permissions.

That said I would want Vercel to be able to do it's thing without yet another research project to figure out settings. I think vendor integration like this is OK. It is a very limited set of actors for GitHub to track and trust.
